packages/ckeditor5-email/tests/emailintegrationutils.js

```diff
@@ -0,0 +1,291 @@
+				{ color: 'hsla(0,0,0,1)', label: 'Black' }
+			] );
+
+			utils._validateConfigColorValue( 'test.colors' );
+
+			sinon.assert.calledWithMatch( warnStub, 'email-integration-unsupported-color-value', {
+				color: 'hsla(0,100%,50%,1)',
+				configPath: 'test.colors[0]'
+			} );
+
+			sinon.assert.calledWithMatch( warnStub, 'email-integration-unsupported-color-value', {
+				color: 'hsla(0,0,0,1)',
+				configPath: 'test.colors[1]'
+			} );
+		} );
+
+		it( 'should not log warning for non-existent config path', () => {
+			const utils = editor.plugins.get( EmailIntegrationUtils );
+
+			utils._validateConfigColorValue( 'non.existent.path' );
+
+			sinon.assert.notCalled( warnStub );
+		} );
+	} );
+
+	describe( '_validateConfigColorFormat()', () => {
+		it( 'should not log warning for rgb format', () => {
+			const utils = editor.plugins.get( EmailIntegrationUtils );
+			editor.config.set( 'test.colorFormat', 'rgb' );
+
+			utils._validateConfigColorFormat( 'test.colorFormat' );
+
+			sinon.assert.notCalled( warnStub );
+		} );
+
+		it( 'should log warning for hsl format', () => {
+			const utils = editor.plugins.get( EmailIntegrationUtils );
+			editor.config.set( 'test.colorFormat', 'hsl' );
+
+			utils._validateConfigColorFormat( 'test.colorFormat' );
+
+			sinon.assert.calledWithMatch( warnStub, 'email-integration-unsupported-color-format', {
+				format: 'hsl',
+				configPath: 'test.colorFormat'
+			} );
+		} );
+
+		it( 'should log warning for hsla format', () => {
+			const utils = editor.plugins.get( EmailIntegrationUtils );
+			editor.config.set( 'test.colorFormat', 'hsla' );
+
+			utils._validateConfigColorFormat( 'test.colorFormat' );
+
+			sinon.assert.calledWithMatch( warnStub, 'email-integration-unsupported-color-format', {
+				format: 'hsla',
+				configPath: 'test.colorFormat'
+			} );
+		} );
+
+		it( 'should not log warning for non-existent config path', () => {
+			const utils = editor.plugins.get( EmailIntegrationUtils );
+
+			utils._validateConfigColorFormat( 'non.existent.path' );
+
+			sinon.assert.notCalled( warnStub );
+		} );
+	} );
+} );
+
+describe( 'isUnsupportedEmailColor()', () => {
+	it( 'should return true for HSL colors', () => {
+		expect( isUnsupportedEmailColor( 'hsl(0,100%,50%)' ) ).to.be.true;
+	} );
+
+	it( 'should return true for HSLA colors', () => {
+		expect( isUnsupportedEmailColor( 'hsla(0,100%,50%,1)' ) ).to.be.true;
+	} );
+
+	it( 'should return false for RGB colors', () => {
+		expect( isUnsupportedEmailColor( 'rgb(255,0,0)' ) ).to.be.false;
+	} );
+
+	it( 'should return false for hex colors', () => {
+		expect( isUnsupportedEmailColor( '#FF0000' ) ).to.be.false;
+	} );
+
+	it( 'should return true for LCH colors', () => {
+		expect( isUnsupportedEmailColor( 'lch(0 100 50)' ) ).to.be.true;
+	} );
+
+	it( 'should return true for LAB colors', () => {
+		expect( isUnsupportedEmailColor( 'lab(0 100 50)' ) ).to.be.true;
+	} );
+
+	describe( 'shortcut', () => {
```

Comment:
This isn't very descriptive.

Reply:
Fixed in https://github.com/ckeditor/ckeditor5/pull/17849/commits/ed892d69946da57dea18b7ee3cebd736a3af3025, separated into two functions.